### ISSUE TITLE
OpenAPI schema generation is missing operationId properties

### ISSUE THREAD

**Issue by Lucidiot** [encode/django-rest-framework#6542]
This issue occurs when using the updated OpenAPI schema generation by @carltongibson (#6532).

Most OpenAPI client tools, such as `apistar` or `openapi3`, require an `operationId` property to be able to perform requests. In the case of APIStar, if the operation ID is missing, a slugified title is used ([code here](https://github.com/encode/apistar/blob/master/apistar/schemas/openapi.py#L420)); however the `title` property seems to also be missing. For those reasons, opening a generated schema with a client will show zero available endpoints.

## Checklist

- [ ] I have verified that that issue exists against the `master` branch of Django REST framework.
- [x] I have searched for similar issues in both open and closed tickets and cannot find a duplicate.
- [x] This is not a usage question. (Those should be directed to the [discussion group](https://groups.google.com/forum/#!forum/django-rest-framework) instead.)
- [x] This cannot be dealt with as a third party library. (We prefer new functionality to be [in the form of third party libraries](https://www.django-rest-framework.org/community/third-party-packages/#about-third-party-packages) where possible.)
- [x] I have reduced the issue to the simplest possible case.
- [ ] I have included a failing test as a pull request. (If you are unable to do so we can still accept the issue.)

## Steps to reproduce

Run `./manage.py generateschema` on any project that has an API endpoint, then load the generated YAML schema with a client library such as API Star:

``` python
>>> import apistar
>>> client = apistar.Client('schema.yaml')
>>> client.document.get_links()
[]
```

APIStar will state that the schema is valid, because `operationId` is not required but recommended, but it will be empty.

## Expected behavior

An OpenAPI schema with `operationId` properties to give a name to each endpoint. It seems the renderer on 3.9/master [did set the property](https://github.com/encode/django-rest-framework/blob/ac19c695396cdd0a72b81dc5fece7a850ae3d1b2/tests/test_generateschema.py#L71).

## Actual behavior

Each path in the schema is missing a name, description, or any kind of identifier that could be used by client libraries.

**Comment by carltongibson** [comment]
Yep. On the list, but input there welcome too. (Both on operationId and title.) 
(The CoreAPI generation [already generates similar](https://github.com/encode/django-rest-framework/blob/ac19c695396cdd0a72b81dc5fece7a850ae3d1b2/rest_framework/schemas/generators.py#L412) so it should just be a case of moving parts around...)

**Comment by Lucidiot** [comment]
@carltongibson I am trying to look into the code to understand how I could experiment with the ID generation, and I was wondering: if `coreapi` is being deprecated, why is it still in use everywhere in the schema generators? Is there a transition planned later on to another library?

**Comment by carltongibson** [comment]
> ...why is it still in use everywhere in the OpenAPI schema generators?

It's not. 🙂 It's still in the **files** because A) it's being deprecated not removed instantly and B) I've not got to the cleanup phase where it might (say) get moved in a separate file. 

If you want to add this... [`OpenAPIAutoSchema.get_operation()`](https://github.com/encode/django-rest-framework/pull/6532/files#diff-9499f248d0768f4456fdb81d78137783R515) : the return dict there needs an extra key. (`operationId`). It'll need to be named off of the view of serializer name, plus a mapping from `method` to e.g. `create`, `destroy` etc. 

If you just want to have a first go at that and open a PR, I'm happy to advise... 🙂

**Comment by carltongibson** [comment]
Closed by #6549
